src/main/java/hudson/tasks/test/TestResult.java

```diff
@@ -276,19 +278,25 @@ public String toPrettyString() {
     }
 
     /**
-     * Annotate some text -- what does this do? 
+     * Annotate text by turning it into a clickable link and then sanitizing it using owasp-java-html-sanitizer
+     * @see <a href="https://www.javadoc.io/doc/com.googlecode.owasp-java-html-sanitizer/owasp-java-html-sanitizer/20160628.1/org/owasp/html/HtmlPolicyBuilder.html">owasp HtmlPolicyBuilder javadoc</a>
      * @param text Text to use to annotate the actions.
      *
      * @return the provided text HTML-escaped.
      */
     public String annotate(String text) {
         if (text == null)
                 return null;
-        text = text.replace("&", "&amp;").replace("<", "&lt;");
 
-        for (TestAction action: getTestActions()) {
-                text = action.annotate(text);
-        }
-        return text;
+        text = text.replace("&", "&amp;").replace("<", "&lt;").replaceAll("\\b(https?://[^\\s)>]+)", "<a href=\"$1\">$1</a>");
+
+        PolicyFactory policy = new HtmlPolicyBuilder()
```

Comment:
`POlicyFactory` is marked as `ThreadSafe` do we really want to create a new one every time this is called?

Reply:
Done, I have moved the policy factory outside of the method so there is only 1 created for the class